### PR TITLE
tidb, metrics: update metrics for session.

### PR DESCRIPTION
**Pull request by coocood**
Move metrics to `metrics` package, and two more, fix retry count.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@tiancaiamao PTAL

File: session.go [hunk 1]
```diff
@@ -724,7 +730,7 @@ func (s *session) executeStatement(goCtx goctx.Context, connID uint64, stmtNode
 		}
 		return nil, errors.Trace(err)
 	}
-	sessionExecuteRunDuration.Observe(time.Since(startTS).Seconds())
```
Comment (zimulala) [inline session.go]: s/startTS/startTime ?

**Comment by coocood** [comment]
@zimulala PTAL

**Pull request merged**